### PR TITLE
Ng2 webpack migration

### PR DESCRIPTION
**Pull request by deepu105**
This is from #4422 which has been merged into below branch
https://github.com/jhipster/generator-jhipster/tree/ng2-webpack

### CODE REVIEW

**Comment by deepu105** [comment]
@jhipster/developers can someone test this branch. with all default option except angular im getting errors below. I'm on linux
```
ERROR in ./src/main/webapp/content/images/logo-jhipster.png
Module build failed: Error: /home/deepu/workspace/temp/jh-webpack/node_modules/pngquant-bin/vendor/pngquant: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory

    at Promise.all.then.arr (/home/deepu/workspace/temp/jh-webpack/node_modules/execa/index.js:170:11)
    at process._tickCallback (internal/process/next_tick.js:103:7)
 @ ./~/css-loader!./src/main/webapp/content/css/main.css 6:2439-2477
 @ ./src/main/webapp/content/css/main.css
 @ ./src/main/webapp/app/vendor.ts

ERROR in ./src/main/webapp/content/images/hipster2x.png
Module build failed: Error: /home/deepu/workspace/temp/jh-webpack/node_modules/pngquant-bin/vendor/pngquant: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory

    at Promise.all.then.arr (/home/deepu/workspace/temp/jh-webpack/node_modules/execa/index.js:170:11)
    at process._tickCallback (internal/process/next_tick.js:103:7)
 @ ./~/css-loader!./src/main/webapp/content/css/main.css 6:3803-3837
 @ ./src/main/webapp/content/css/main.css
 @ ./src/main/webapp/app/vendor.ts

ERROR in ./src/main/webapp/content/images/hipster.png
Module build failed: Error: /home/deepu/workspace/temp/jh-webpack/node_modules/pngquant-bin/vendor/pngquant: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory

    at Promise.all.then.arr (/home/deepu/workspace/temp/jh-webpack/node_modules/execa/index.js:170:11)
    at process._tickCallback (internal/process/next_tick.js:103:7)
 @ ./~/css-loader!./src/main/webapp/content/css/main.css 6:3225-3257
 @ ./src/main/webapp/content/css/main.css
 @ ./src/main/webapp/app/vendor.ts

ERROR in ./src/main/webapp/content/css/main.css
Module build failed: ModuleBuildError: Module build failed: Error: /home/deepu/workspace/temp/jh-webpack/node_modules/pngquant-bin/vendor/pngquant: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory

    at Promise.all.then.arr (/home/deepu/workspace/temp/jh-webpack/node_modules/execa/index.js:170:11)
    at process._tickCallback (internal/process/next_tick.js:103:7)
    at /home/deepu/workspace/temp/jh-webpack/node_modules/webpack/lib/NormalModule.js:143:35
    at /home/deepu/workspace/temp/jh-webpack/node_modules/loader-runner/lib/LoaderRunner.js:359:11
    at /home/deepu/workspace/temp/jh-webpack/node_modules/loader-runner/lib/LoaderRunner.js:225:18
    at context.callback (/home/deepu/workspace/temp/jh-webpack/node_modules/loader-runner/lib/LoaderRunner.js:106:13)
    at imagemin.buffer.then.catch.err (/home/deepu/workspace/temp/jh-webpack/node_modules/image-webpack-loader/index.js:48:9)

ERROR in /home/deepu/workspace/temp/jh-webpack/node_modules/extract-text-webpack-plugin/loader.js?{"omit":1,"remove":true}!/home/deepu/workspace/temp/jh-webpack/node_modules/style-loader/index.js!/home/deepu/workspace/temp/jh-webpack/node_modules/css-loader/index.js!/home/deepu/workspace/temp/jh-webpack/src/main/webapp/content/css/main.css doesn't export content

ERROR in [default] /home/deepu/workspace/temp/jh-webpack/src/main/webapp/app/blocks/config/localstorage.config.ts:1:9 
Module '"/home/deepu/workspace/temp/jh-webpack/node_modules/ng2-webstorage/dist/app"' has no exported member 'KeyStorageHelper'.

ERROR in [default] /home/deepu/workspace/temp/jh-webpack/src/main/webapp/app/shared/language/jhi-missing-translation.config.ts:3:13 
Class 'JhiMissingTranslationHandler' incorrectly implements interface 'MissingTranslationHandler'.
  Types of property 'handle' are incompatible.
    Type '(key: string) => string' is not assignable to type '(params: MissingTranslationHandlerParams) => any'.
      Types of parameters 'key' and 'params' are incompatible.
        Type 'MissingTranslationHandlerParams' is not assignable to type 'string'.
Child html-webpack-plugin for "index.html":
        + 21 hidden modules
Child extract-text-webpack-plugin:
        + 7 hidden modules
Child extract-text-webpack-plugin:
        + 5 hidden modules
    
    ERROR in ./src/main/webapp/content/images/logo-jhipster.png
    Module build failed: Error: /home/deepu/workspace/temp/jh-webpack/node_modules/pngquant-bin/vendor/pngquant: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
    
        at Promise.all.then.arr (/home/deepu/workspace/temp/jh-webpack/node_modules/execa/index.js:170:11)
        at process._tickCallback (internal/process/next_tick.js:103:7)
     @ ./~/css-loader!./src/main/webapp/content/css/main.css 6:2439-2477
    
    ERROR in ./src/main/webapp/content/images/hipster.png
    Module build failed: Error: /home/deepu/workspace/temp/jh-webpack/node_modules/pngquant-bin/vendor/pngquant: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
    
        at Promise.all.then.arr (/home/deepu/workspace/temp/jh-webpack/node_modules/execa/index.js:170:11)
        at process._tickCallback (internal/process/next_tick.js:103:7)
     @ ./~/css-loader!./src/main/webapp/content/css/main.css 6:3225-3257
    
    ERROR in ./src/main/webapp/content/images/hipster2x.png
    Module build failed: Error: /home/deepu/workspace/temp/jh-webpack/node_modules/pngquant-bin/vendor/pngquant: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory
    
        at Promise.all.then.arr (/home/deepu/workspace/temp/jh-webpack/node_modules/execa/index.js:170:11)
        at process._tickCallback (internal/process/next_tick.js:103:7)
     @ ./~/css-loader!./src/main/webapp/content/css/main.css 6:3803-3837

```

**Comment by mraible** [comment]
I'm getting the following error when trying to use this branch with `yo jhipster`. It doesn't even make it past the install process.

```
identical src/main/resources/i18n/messages_en.properties
events.js:141
      throw er; // Unhandled 'error' event
      ^

TypeError: paths.split is not a function
    at install.runInstall (/Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/lib/actions/install.js:33:57)
    at install [as npmInstall] (/Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/lib/actions/install.js:160:15)
    at module.exports.JhipsterClientGenerator.extend.install (/Users/mraible/dev/generator-jhipster/generators/client-2/index.js:293:18)
    at Object.<anonymous> (/Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/lib/base.js:439:23)
    at /Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/node_modules/run-async/index.js:26:25
    at /Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/node_modules/run-async/index.js:25:19
    at /Users/mraible/dev/generator-jhipster/node_modules/yeoman-generator/lib/base.js:440:9
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

**Comment by mraible** [comment]
I tried with Node v6.9.1 and the same issue happens.

**Comment by pascalgrimaud** [comment]
1) launch `yo jhipster`
2) default option except angular 2.x
3) error (same as @mraible) :
```
   create src/main/resources/i18n/messages_en.properties
events.js:160
      throw er; // Unhandled 'error' event
      ^

TypeError: paths.split is not a function
    at constructor.install.runInstall (/home/pgrimaud/projects/jhipster/generator-jhipster/node_modules/yeoman-generator/lib/actions/install.js:33:57)
    at constructor.install [as npmInstall] (/home/pgrimaud/projects/jhipster/generator-jhipster/node_modules/yeoman-generator/lib/actions/install.js:160:15)
    at constructor.install (/home/pgrimaud/projects/jhipster/generator-jhipster/generators/client-2/index.js:293:18)
    at Object.<anonymous> (/home/pgrimaud/projects/jhipster/generator-jhipster/node_modules/yeoman-generator/lib/base.js:439:23)
    at /home/pgrimaud/projects/jhipster/generator-jhipster/node_modules/run-async/index.js:26:25
    at /home/pgrimaud/projects/jhipster/generator-jhipster/node_modules/run-async/index.js:25:19
    at /home/pgrimaud/projects/jhipster/generator-jhipster/node_modules/yeoman-generator/lib/base.js:440:9
    at runCallback (timers.js:637:20)
    at tryOnImmediate (timers.js:610:5)
    at processImmediate [as _immediateCallback] (timers.js:582:5)
```

1) launch `yo jhipster --skip-install`
2) default option except angular 2.x
3) everything is generated. I launch `npm install`
I got some error in the tree:
```
├─┬ angular2-cookie@1.2.5 
│ ├── UNMET PEER DEPENDENCY rxjs@5.0.0-beta.12
│ └── UNMET PEER DEPENDENCY zone.js@0.6.26

├─┬ UNMET PEER DEPENDENCY rxjs@5.0.0-rc.4

├─┬ UNMET PEER DEPENDENCY webpack@2.1.0-beta.25

└── UNMET PEER DEPENDENCY zone.js@0.7.2

```

4) then, `./mvnw` or `npm run webpack-dev` I got error:

```
ERROR in [default] /home/pgrimaud/test/05-ng2/src/main/webapp/app/blocks/config/localstorage.config.ts:1:9 
Module '"/home/pgrimaud/test/05-ng2/node_modules/ng2-webstorage/dist/app"' has no exported member 'KeyStorageHelper'.

ERROR in [default] /home/pgrimaud/test/05-ng2/src/main/webapp/app/shared/language/jhi-missing-translation.config.ts:3:13 
Class 'JhiMissingTranslationHandler' incorrectly implements interface 'MissingTranslationHandler'.
  Types of property 'handle' are incompatible.
    Type '(key: string) => string' is not assignable to type '(params: MissingTranslationHandlerParams) => any'.
      Types of parameters 'key' and 'params' are incompatible.
        Type 'MissingTranslationHandlerParams' is not assignable to type 'string'.
```

**Comment by gmarziou** [comment]
I got same error as Matt on Windows 7 and nodejs v4.3.1

**Comment by gmarziou** [comment]
I found a message that I did not notice first: `errorSyntaxError: Unexpected token <`

~~~
This is an existing project, using the configuration from your .yo-rc.json file
to re-generate the project...

 ______________________________________________________________________________

  JHipster update available: 3.11.0 (current: 3.9.1)
  Run npm install -g generator-jhipster to update.
 ______________________________________________________________________________

errorSyntaxError: Unexpected token <
identical package.json
identical README.md
identical .gitignore

~~~

**Comment by deepu105** [comment]
I might have introduced the error during install. But the remaining is
already there I guess

Thanks & regards,
Deepu

**Comment by deepu105** [comment]
Ok this is annoying I'm gonna merge this since angular2 branch is broken anyway

**Comment by flaviencathala** [comment]
@deepu105 You can fiw the paths.split error by changing `this.npmInstall` in `this.installDependencies` in client-2/index.js line 293.

**Comment by flaviencathala** [comment]
@deepu105 I would have like to fix it myself but I don't have the rights.

**Comment by deepu105** [comment]
Its merged to angular2 now so you can do PR to that with fixes.

Thanks & regards,
Deepu

**Comment by flaviencathala** [comment]
@deepu105 Are you sure it's merged ? I don't see the ng2-webpack files in angular-2 branch.

**Comment by deepu105** [comment]
check now, i also fixed the install bug

**Comment by deepu105** [comment]
i forgot to push it actually

**Comment by flaviencathala** [comment]
@deepu105 Okay thanks.

File: generators/client-2/templates/_webpack-dev [hunk 1]
```diff
@@ -0,0 +1 @@
+npm run webpack-dev
```
Comment (deepu105) [inline generators/client-2/templates/_webpack-dev]: @flaviencathala i think we can remove the file as its clear in the package.json and we now print the info during generation as well so thats sufficient

Comment (flaviencathala) [inline generators/client-2/templates/_webpack-dev]: Okay as you want.

File: generators/client-2/templates/config/_webpack.common.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+const webpack = require('webpack');
```
Comment (deepu105) [inline generators/client-2/templates/config/_webpack.common.js]: @flaviencathala @jdubois can we rename the folder from `config` to `webpack`?

Comment (flaviencathala) [inline generators/client-2/templates/config/_webpack.common.js]: It's ok for us.

**Pull request merged**